### ISSUE TITLE
HDT parse support

### ISSUE THREAD

**Issue by KonradHoeffner** [oxigraph/rio#101]
Given that they are much smaller than other serialization formats, it would be very useful to load [HDT](https://www.rdfhdt.org/) (Header, Dictionary, Triples) files with rio.
There seem to be only C++ and  Java libraries that handle HDT right now, so having this in the Rust world would allow users to create even more performant RDF applications with Rust.

**Comment by Tpt** [comment]
Yes, it would be great indeed. But to provide efficient access to the HDT file the reader should expose a quite different API from the one of the other RDF format parsers.

**Comment by KonradHoeffner** [comment]
In that case it would maybe make sense to create a separate HDT crate. I will give it a try, as there seems to exist no such thing at the moment, at least I didn't find anything on crates.io and didn't receive any answers at <https://softwarerecs.stackexchange.com/questions/83973/rdf-hdt-header-dictionary-triples-library-in-rust>.

**Comment by KonradHoeffner** [comment]
Update: I found a prototype at <https://github.com/timplication/hdt-rs/> and extended it at <https://github.com/konradhoeffner/hdt-rs>, maybe this could be useful to you, however it is quite complicated and I can't guarantee if everything is working as intended or performant enough.

**Comment by Tpt** [comment]
@KonradHoeffner Thank you so much for this work!

**Comment by KonradHoeffner** [comment]
Thanks! The vast majority was done by @timplication though, I'm just trying to add the things that are missing :-)
